### DISCUSSION TITLE
Can’t seem to get LED Feedback for MIDI to work properly on my MiniLab 3 using ReaLearn

### DISCUSSION THREAD

**Discussion by gentrophonic** [helgoboss/helgobox#1260] [Q&A]
Hello everyone-

All I want is for the pads on my MiniLab 3 to show me when a track is unmuted/muted. This seems like one of the easiest things to do. And maybe it is—but I’m so deeply down a path of complete idiocy that I’m missing something. I wonder if u/helgoboss can maybe chime in (crosses fingers).

I’m trying to put together my first live performance using Playtime as a clip launcher, Super 8 as a looper, and ReaLearn as the main midi mapping protocol. So, far, despite the wonkiness of it all, it has been rather straightforward to program and perform. If I can just get this LED feedback issue fixed, it will make things so much easier.

 I can't get my LED feedback on the MIDI pads from MiniLab 3 how I want them. Maybe if someone could look at the screen shot and video I attached you might be able to help. As I explain in the video, the LED lights brighten when the track is muted AND unmuted; thus giving me no indication as to the state of the track without either remembering what I pushed last, looking closely at the tracks/mixer in the .rpp session, or simply listening. 

What I've done so far:
1) I feel like I've got through every relevant permutation in the ReaLearn mapping menu. I've messed around with the source, target, and glue. I've done it so much and so many variations, I lost track as to what was happening in one track mapping, that I had to delete it altogether because it became non-functional.
   1) Specifically working in the “for buttons” menu within target feedback settings in the glue panel.
2) Looked up tutorials RE: ReaLearn and LED feedback. Most of which discuss dealing with changing encoder colors and directions
3) Looked through the  ReaLearn User Guide--but maybe not deep enough. It is an academically dense read.
4) Went to MCC app through Arturia to create a user setting wtih modified LED colors. Tried the pads in Toggle and Gate mode; either mode doesn’t change the behavior of the ReaLearn mapping algorithm.

What logical reasoning is telling me should be the answer:
1) Using MIDI control message—> note on=unmuted track and where note off=muted track
   1) Note On=LED faint
   2) Note Off=LED bright
2) It seems like it has something to do with the “glue” panel of the ReaLearn mapping window, specifically in the target feedback section. 
3) Maybe there is a special script out there?

Screenshot of current ReaLearn mapping 
![Screenshot 2024-10-05 at 12 18 56 PM](https://github.com/user-attachments/assets/3ad1893e-f46e-4e40-9af7-c5f411148f14)
Quick Video that shows the issue
[https://www.dropbox.com/scl/fi/wlck1r6ze39fpfux7ae9t/minilab3_ReaLearn_LED_Mapping.mov?rlkey=1w6ps83ug8l34n39n5g4sze24&st=nzitnor0&dl=0](url)

**Comment by helgoboss** [comment]
I think i answered this in the forum. Please don't double post.

**Comment by gentrophonic** [comment]
My apologies. I think I figured it out. Christian Gentryhttp://christiangentry.comOn Oct 19, 2024, at 2:23 AM, Benjamin Klum ***@***.***> wrote:﻿
I think i answered this in the forum. Please don't double post.

**Reply by helgoboss** [reply]
All good. Glad you figured it out.
